internal/sync.go

```diff
@@ -57,24 +57,34 @@ func (s *syncGSuite) SyncUsers() error {
 	log.Debug("get deleted users")
 	deletedUsers, err := s.google.GetDeletedUsers()
 	if err != nil {
+		log.Warn("Error Getting Deleted Users")
 		return err
 	}
 
 	for _, u := range deletedUsers {
+		log.WithFields(log.Fields{
+			"email": u.PrimaryEmail,
+		}).Info("deleting google user")
+
 		uu, err := s.aws.FindUserByEmail(u.PrimaryEmail)
-		if err != aws.ErrUserNotFound {
+		if err != aws.ErrUserNotFound && err != nil {
```

Comment:
Let's make this simpler. `if uu == nil { return err }`. it should always be nil when there is and error.

Reply:
I think not. for the following use-case
deleted_users = [userA@domain.com, userB@domain.com]
UserA is doesn't exist but userB yes on SSO

FinduserByEmail (userA)
* err =  aws.ErrUserNotFound
* uu = nil
action -> **continue** 
go to the next user in the list

FinduserByEmail (userB)
* err = nil
* uu = <object>
action 
-> DeleteUser
go to the next user in the list

if uu == nil and the error is different , then the function should break the whole loop and return err as there is an unexpected errror